src/Compilers/CSharp/Portable/Symbols/MemberSymbolExtensions.cs

```diff
@@ -658,5 +658,16 @@ internal static string GetMemberCallerName(this Symbol member)
                 member.IsExplicitInterfaceImplementation() ? ExplicitInterfaceHelpers.GetMemberNameWithoutInterfaceName(member.Name) :
                 member.Name;
         }
+
+        internal static bool CanHaveFieldKeywordBackingField(this Symbol symbol)
+        {
+            Debug.Assert(symbol.ContainingSymbol is SourceMemberContainerTypeSymbol or SourceNamespaceSymbol or SourceModuleSymbol);
```

Comment:
>SourceMemberContainerTypeSymbol or SourceNamespaceSymbol or SourceModuleSymbol [](http://example.com/codeflow?start=52&length=78)

I think this is an unnecessary strong and probably is going to fail in EE. I suggest to check symbol.Kind instead and remove the next assert as well.

Reply:
Fixed (I assume you meant `symbol.ContainingSymbol.Kind`)